types/lodash/lodash-tests.ts

```diff
@@ -4326,6 +4326,20 @@ fp.now(); // $ExpectType number
 
 // _.isFinite
 {
+    const value: string | number = 0;
+
+    if (_.isFinite(value)) {
+        const result: number = value;
+    } else {
+        const result: string = value;
```

Comment:
This is wrong. If `isFinite` returns false, then `value` isn't necessarily a string. It could be a number (`Infinity` or `NaN`).

Reply:
Ooh, yeah, the negative case breaks this all apart. Guess it's not possible. :(